### PR TITLE
DTrace probes for upstairs work queue

### PR DESCRIPTION
**Pull request by leftwo**
Added internal counters for job states in the upstairs and hooked those into the dtrace probes we already had.

Updated the dtrace info script to print out these counters.

### CODE REVIEW

**Comment by faithanalog** [comment]
Could you put an example invocation of using `tools/dtrace/upstairs_info.d` with dtrace on a running upstairs, maybe as a comment in the `upstairs_info.d` file itself or wherever is convenient? For the benefit of us less familiar with using dtrace.

**Comment by faithanalog** [comment]
oh i didnt even know that README file existed, dont i feel like a fool now!

**Comment by leftwo** [comment]
> oh i didnt even know that README file existed

It needed updating, so thanks for reminding me about it.
I'm actually going to add a bit more to it to explain better what
the numbers mean.

**Comment by leftwo** [comment]
Okay, I'm done with polishing.
Hopefully these updates help address any concerns.

File: tools/dtrace/README.md [hunk 1]
```diff
@@ -263,24 +263,53 @@ Trace a downstairs IO and measure time for in in the following three parts:
...
 If the upstairs is not yet running, add the -Z flag to dtrace so it will
 wait to find the matching probe.
 ```
 pfexec dtrace -s upstairs_info.d
 ```
 
 You start crucible, then run the above script.  Output should start appearing
-right away with the state of the three downstairs and a count of active
-jobs for upstairs and downstairs.
+with a few seconds.
```
Comment (faithanalog) [inline tools/dtrace/README.md:274]: extreme nit moment: probably "within"

File: tools/dtrace/README.md [hunk 2]
```diff
@@ -263,24 +263,53 @@ Trace a downstairs IO and measure time for in in the following three parts:
...
+
+Now, back to the columns.
+Columns four and five show the count of active jobs for the guest side (UPW)
+and for the downstairs side (DSW).
+
+
+The remaining columns 6-20 show the count of job states on each downstairs.
+Jobs for the downstairs can be in one of five possible states:
+ `New`, `In Progress`, `Done`, `Skipped`, `Error`.
+For each state/downstairs, we print a count of jobs in that state.
```
Comment (faithanalog) [inline tools/dtrace/README.md:300]: this is very helpful thanks!!

**Review by faithanalog — APPROVED** [review]
thanks for the readme updates! i left a nit on something that might be a typo, otherwise, ship it imo

File: tools/dtrace/README.md (hunk 1 above)
Comment (leftwo) [reply tools/dtrace/README.md:274]: Neither extreme nor a nit, that's a straight up error!

**Pull request merged**